File: python/66_Plus_One.py
class Solution:
    # @param {integer[]} digits
    # @return {integer[]}
    def plusOne(self, digits):
        digits[-1]=digits[-1]+1
        for index in range(len(digits)-1,-1,-1):
            c=digits[index]/10
            if c!=0:
                digits[index]=digits[index]%10
                if index>0:
                    digits[index-1]+=c
                else:
                    digits=[c]+digits
        return digits

File: Java/66_Plus_One.java
public class Solution {
    public int[] plusOne(int[] digits) {
        // Easy题，没坑，注意进位
        if(digits==null||digits.length==0)return digits;
        int len=digits.length;
        int i=len-2, c=(digits[len-1]+1)/10;
        digits[len-1] = (digits[len-1]+1)%10;
        while(c>0&&i>=0){
            digits[i]+=c;
            c=digits[i]/10;
            digits[i]=digits[i]%10;
            i--;
        }
        if(c>0){
            int[] news = new int[len+1];
            news[0]=c;
            for(i=0;i<len;i++){
                news[i+1]=digits[i];
            }
            return news;
        }
        return digits;
    }
}